plugins/cost-insights/src/components/EntityCosts/EntityCosts.tsx

```diff
@@ -0,0 +1,160 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useCallback, useEffect, useState } from 'react';
+import { CostOverviewCard } from '../CostOverviewCard';
+import {
+  BillingDateProvider,
+  ConfigProvider,
+  CurrencyProvider,
+  FilterProvider,
+  GroupsProvider,
+  LoadingProvider,
+  ScrollProvider,
+  useFilters,
+  useLastCompleteBillingDate,
+  useLoading,
+} from '../../hooks';
+import { CostInsightsThemeProvider } from '../CostInsightsPage/CostInsightsThemeProvider';
+import { Progress, WarningPanel } from '@backstage/core-components';
+import { default as MaterialAlert } from '@material-ui/lab/Alert';
+import { mapLoadingToProps } from '../CostInsightsPage/selector';
+import { intervalsOf } from '../../utils/duration';
+import { costInsightsApiRef } from '../../api';
+import { useApi } from '@backstage/core-plugin-api';
+import { useEntity } from '@backstage/plugin-catalog-react';
+import { Cost, Maybe } from '@backstage/plugin-cost-insights-common';
+
+export const EntityCostsCard = () => {
+  const client = useApi(costInsightsApiRef);
+  const { entity } = useEntity();
+
+  const {
+    loadingActions,
+    loadingGroups,
+    loadingBillingDate,
+    loadingInitial,
+    dispatchInitial,
+    dispatchInsights,
+    dispatchNone,
+  } = useLoading(mapLoadingToProps);
+
+  /* eslint-disable react-hooks/exhaustive-deps */
+  // The dispatchLoading functions are derived from loading state using mapLoadingToProps, to
+  // provide nicer props for the component. These are re-derived whenever loading state changes,
+  // which causes an infinite loop as product panels load and re-trigger the useEffect below.
+  // Since the functions don't change, we can memoize - but we trigger the same loop if we satisfy
+  // exhaustive-deps by including the function itself in dependencies.
+
+  const dispatchLoadingInitial = useCallback(dispatchInitial, []);
+  const dispatchLoadingInsights = useCallback(dispatchInsights, []);
+  const dispatchLoadingNone = useCallback(dispatchNone, []);
+  /* eslint-enable react-hooks/exhaustive-deps */
+
+  const lastCompleteBillingDate = useLastCompleteBillingDate();
+  const [dailyCost, setDailyCost] = useState<Maybe<Cost>>(null);
+  const [error, setError] = useState<Maybe<Error>>(null);
+  const { pageFilters } = useFilters(p => p);
+
+  useEffect(() => {
+    async function getInsights() {
+      setError(null);
+      try {
+        dispatchLoadingInsights(true);
+        const intervals = intervalsOf(
+          pageFilters.duration,
+          lastCompleteBillingDate,
+        );
+
+        const fetchedDailyCost = await client.getEntityDailyCost!(
```

Comment:
I assume we have to check for the existence of this method before :)

Reply:
This check is performed in the `EntityCosts` component, which is the one that ends up being exposed publicly by this plugin. Should I replicate the same check in this component too?